File: javascript/lesson 4/countingSort.js
function countingSort(nums, max) {
    const freq = new Array(max + 1).fill(0);
    nums.forEach(num => {
        freq[num]++;
    });

    let current = 0;
    for (let i = 0; i < freq.length; i++) {
        for (let count = 0; count < freq[i]; count++) {
            nums[current] = i;
            current += 1;
        }
    }
}

File: csharp/lesson 4/CountingSort.cs
public class CountingSort 
{
    void CountingSort(int[] nums, int max) {
        var freq = new int[max + 1];
        foreach (var num in nums)
        {
            freq[num]++;
        }
        
        var current = 0;
        for (var i = 0; i < freq.Length; i++)
        {
            for (var count = 0; count < freq[i]; count++)
            {
                nums[current] = i;
                current++;
            }
        }
    }
}
